worlds/dkc3/Regions.py

```diff
@@ -0,0 +1,882 @@
+        })
+
+        bramble_region_locations.update({
+            LocationName.brambles_bungalow: [0x619, 2],
+        })
+
+        #flower_spot_region_locations.update({
+        #    LocationName.flower_spot: [0x615, 3, True],
+        #})
+
+        barter_region_locations.update({
+            LocationName.barters_swap_shop: [0x61B, 3],
+        })
+
+        barnacle_region_locations.update({
+            LocationName.barnacles_island: [0x61D, 2],
+        })
+
+        blue_region_locations.update({
+            LocationName.blues_beach_hut: [0x621, 4],
+        })
+
+        blizzard_region_locations.update({
+            LocationName.blizzards_basecamp: [0x625, 4, True],
+        })
+
+    bazaar_region = create_region(world, player, active_locations, LocationName.bazaar_region,
+                                  bazaar_region_locations, None)
+    bramble_region = create_region(world, player, active_locations, LocationName.bramble_region,
+                                   bramble_region_locations, None)
+    flower_spot_region = create_region(world, player, active_locations, LocationName.flower_spot_region,
+                                       flower_spot_region_locations, None)
+    barter_region = create_region(world, player, active_locations, LocationName.barter_region,
+                                  barter_region_locations, None)
+    barnacle_region = create_region(world, player, active_locations, LocationName.barnacle_region,
+                                    barnacle_region_locations, None)
+    blue_region = create_region(world, player, active_locations, LocationName.blue_region,
+                                blue_region_locations, None)
+    blizzard_region = create_region(world, player, active_locations, LocationName.blizzard_region,
+                                    blizzard_region_locations, None)
+
+    world.regions += [
+        bazaar_region,
+        bramble_region,
+        flower_spot_region,
+        barter_region,
+        barnacle_region,
+        blue_region,
+        blizzard_region,
+    ]
+
+
+def connect_regions(world, player, level_list):
+    names: typing.Dict[str, int] = {}
+
+    # Overworld
+    connect(world, player, names, 'Menu', LocationName.overworld_1_region)
+    connect(world, player, names, LocationName.overworld_1_region, LocationName.overworld_2_region,
+            lambda state: (state.has(ItemName.progressive_boat, player, 1)))
+    connect(world, player, names, LocationName.overworld_2_region, LocationName.overworld_3_region,
+            lambda state: (state.has(ItemName.progressive_boat, player, 3)))
+    connect(world, player, names, LocationName.overworld_1_region, LocationName.overworld_4_region,
+            lambda state: (state.has(ItemName.dk_coin, player, world.dk_coins_for_gyrocopter[player].value)))
+
+    connect(world, player, names, LocationName.overworld_1_region, LocationName.overworld_4_region,
```

Comment:
Is this and the function above it correct? This looks odd to me.

Reply:
Yep, forgot to remove the first one when I added the second one.